### PR TITLE
New Adamantine Effect and Gauntlet Fix

### LINKED ISSUE #102: Adamantine Armor Effect
Adamantine Effect Idea:

Every X tick (this will be configurable) the player will gain a food level by using experience

I've created this effect idea to nerf the current one

### PR DESCRIPTION
**Pull request by PierKnight100**
This is a:
*   Code Pull request

#### Description of the pull request content

This pull request contains:

- The New Effect for the Adamantine Armor [ Fix #102 ]
- A fix: now the gauntlet consumes saturation too #104

---
#### Reason why we should implement your feature into the mod
To nerf the adamantine armor and fix the gauntlet

### CODE REVIEW

**Comment by Klazkin** [comment]
Suggestion:
Instead of making a "set bonus" that only works with full armour - make it so that each armour piece increases the efficiency ratio of XP to hunger conversion by X amount. Minor change, but I think encouraging different armour combinations is a bit more fun.

**Comment by PierKnight100** [comment]
@Davoleo  Done

File: src/main/java/it/hurts/metallurgy_reforged/util/capabilities/punch/PunchEffect.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ public Vec3d getKnockbackMotionVec() {
 		double y = -MathHelper.sin(pitch * 0.017453292F);		      
 		double z = MathHelper.cos(yaw * 0.017453292F) * MathHelper.cos(pitch * 0.017453292F);
 		double f = MathHelper.sqrt(x * x + y * y + z * z);
-		double velocity = 2D;
+		double velocity = this.getHitTicks() > 10 ? 2D : 1D;
```
Comment (HurtsAsHell) [inline src/main/java/it/hurts/metallurgy_reforged/util/capabilities/punch/PunchEffect.java]: Non ho capito questo if @PierKnight100 puoi spiegarmelo?

**Comment by Davoleo** [comment]
> Suggestion:
> Instead of making a "set bonus" that only works with full armour - make it so that each armour piece increases the efficiency ratio of XP to hunger conversion by X amount. Minor change, but I think encouraging different armour combinations is a bit more fun.

This won't get lost, I'll save it in our archive, thanks for your suggestion @Klazkin =)

**Pull request merged**